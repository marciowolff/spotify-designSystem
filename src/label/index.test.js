import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Label from './index'

it('Label should be a function', () => {
  expect(typeof Label).toBe('function')
})

it('Should Input to match snapshot', () => {
  const tree = renderer.create(<Label htmlFor="teste">texto</Label>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Label should with text', () => {
  const wrapper = shallow(<Label htmlFor="teste">texto</Label>)

  expect(wrapper.text()).toBe('texto')
})

it('Should should without text', () => {
  const wrapper = shallow(<Label htmlFor="teste" />)

  expect(wrapper.text()).toBe('')
})

it('Should should with htmlFor', () => {
  const wrapper = shallow(<Label htmlFor="teste" />)

  expect(wrapper.props().htmlFor).toBe('teste')
})

it('Should should without htmlFor', () => {
  const wrapper = shallow(<Label />)

  expect(wrapper.props().htmlFor).toBeFalsy()
})
