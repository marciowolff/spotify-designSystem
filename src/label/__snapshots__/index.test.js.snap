// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Input to match snapshot 1`] = `
<label
  htmlFor="teste"
>
  texto
</label>
`;
